addOns/fuzz/src/main/java/org/zaproxy/zap/extension/fuzz/FuzzAPI.java

```diff
@@ -0,0 +1,842 @@
+    private static final String VIEW_GET_MESSAGES_SENT = "getMessagesSentCount";
+    // TODO implement this, get a list of all the messages that were sent
+    private static final String VIEW_GET_ALL_SENT_MESSAGES = "getAllSentMessages";
+    private static final String VIEW_GET_RESULTS = "getResults";
+    private static final String ACTION_START_SCAN = "startScan";
+    private static final String ACTION_STOP_SCAN = "stopScan";
+    private static final String ACTION_PAUSE_SCAN = "pauseScan";
+    private static final String ACTION_RESUME_SCAN = "resumeScan";
+
+    /** Provided only for API client generator usage. */
+    public FuzzAPI() {
+        this(null);
+    }
+
+    /* constructor to be used to receive ACTIONS, VIEW requests from the user.*/
+    public FuzzAPI(ExtensionFuzz ext) {
+        httpFuzzerMessageProcessors = new ArrayList<>();
+        this.extension = ext;
+        this.addApiAction(
+                new ApiAction(
+                        ACTION_SIMPLE_HTTP_FUZZER,
+                        new String[] {
+                            PARAM_MESSAGE_ID,
+                            PARAM_FUZZ_REQUEST_LOCATION,
+                            PARAM_FUZZ_LOCATION,
+                            PARAM_PAYLOAD_PATH
+                        }));
+        this.addApiAction(
+                new ApiAction(
+                        ACTION_MULTIPLE_PAYLOAD_FUZZER,
+                        new String[] {
+                            PARAM_MESSAGE_ID, PARAM_JSON_LOCATION,
+                        }));
+        this.addApiAction(
+                new ApiAction(
+                        ACTION_SET_HTTP_FUZZ_OPTIONS,
+                        new String[] {},
+                        new String[] {
+                            PARAM_MAX_ERRORS_ALLOWED,
+                            PARAM_STRATEGY,
+                            PARAM_RETRIES,
+                            PARAM_DELAY,
+                            PARAM_THREADS
+                        }));
+        this.addApiAction(new ApiAction(ACTION_RESET_DEFAULT_HTTP_FUZZ_OPTIONS));
+        this.addApiAction(new ApiAction(ACTION_START_SCAN, new String[] {PARAM_FUZZER_ID}));
+        this.addApiAction(new ApiAction(ACTION_STOP_SCAN, new String[] {PARAM_FUZZER_ID}));
+        this.addApiAction(new ApiAction(ACTION_PAUSE_SCAN, new String[] {PARAM_FUZZER_ID}));
+        this.addApiAction(new ApiAction(ACTION_RESUME_SCAN, new String[] {PARAM_FUZZER_ID}));
+
+        this.addApiView(new ApiView(VIEW_FUZZER_PROGRESS, new String[] {PARAM_FUZZER_ID}));
+        this.addApiView(new ApiView(VIEW_GET_RESULTS, new String[] {PARAM_FUZZER_ID}));
+        this.addApiView(new ApiView(VIEW_GET_MESSAGES_SENT, new String[] {PARAM_FUZZER_ID}));
+        this.addApiView(new ApiView(VIEW_GET_ALL_SENT_MESSAGES, new String[] {PARAM_FUZZER_ID}));
+    }
+
+    @Override
+    public String getPrefix() {
+        return PREFIX;
+    }
+
+    /** resets the api "only" fuzzerOptionsToDefault */
```

Comment:
This comment seems un-necessary.

Though arguably having jdoc is good, if you leave it please format it properly and use title caps appropriately.